cache/disk/lru.go

```diff
@@ -120,6 +124,44 @@ func (c *sizedLRU) MaxSize() int64 {
 	return c.maxSize
 }
 
+func (c *sizedLRU) Reserve(size int64) bool {
+	if size < 0 || size > c.maxSize || c.reservedSize+size > c.maxSize {
+		return false
+	}
+
+	// Evict elements until we are able to reserve enough space.
+	for c.currentSize+size > c.maxSize {
+		ele := c.ll.Back()
+		if ele != nil {
+			c.removeElement(ele)
+		} else {
+			return false // This should have been caught at the start.
```

Comment:
If this happens it's a bug?

Reply:
Yes.

Reply:
Shouldn't we fail somehow then and not just return false? This will put the cache in an undefined state?